shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java

```diff
@@ -80,8 +81,17 @@
     
     private final String schemaName;
 
+    private final Map<String, ShardingSphereMetaData> metaDataMap;
+
+    private final List<Object> parameters;
+
+    @Setter
+    private Map<String, Map<String, Map<String, String>>> rewriteMetaDataMap;
```

Comment:
What is the purpose of adding these variables?

Reply:
to initialize selectStatementContext with subquery selectStatement